pkg/apis/pipeline/v1beta1/resultref.go

```diff
@@ -129,3 +129,27 @@ func parseExpression(substitutionExpression string) (string, string, error) {
 	}
 	return subExpressions[1], subExpressions[3], nil
 }
+
+// PipelineTaskResultRefs walks all the places a result reference can be used
+// in a PipelineTask and returns a list of any references that are found.
+func PipelineTaskResultRefs(pt *PipelineTask) []*ResultRef {
```

Comment:
thanks @sbwsg this is a huge simplification. But I am trying to understand this, any specific reason moving this logic here from `pipeline_types.go`? collecting all resultrefs under `resultref.go`?

Reply:
Are you referring here to the fact that I originally made this a method on `*PipelineTask`?

In a follow-up PR I'd like to move all the result-related code to its own package. Just like we have `pkg/workspace` I think we should have `pkg/results`.  It's always been a little strange to me that all of the results processing lives in `v1beta1`.

So what I'd like to avoid is attaching the result processing logic to our v1beta1 types.  I initially added this as a method on `*PipelineTask` because it was "neat" to be able to call `pt.ResultReferences()`.  But I ultimately wasn't convinced that the "neatness" was worth coupling that logic to the beta type.

Reply:
nevermind got it 😜  its a pure refactoring to minimize iterating over resultRefs in three separate for loops.

Reply:
yup, sounds good @sbwsg thanks for explaining it.